services/QuillLMS/engines/comprehension/app/models/comprehension/rule.rb

```diff
@@ -68,14 +68,21 @@ def determine_feedback_from_history(feedback_history)
 
     def regex_is_passing?(entry)
       regex_rules.none? do |regex_rule|
```

Comment:
If you put the matching function on the `RegexRule` model, this function becomes a lot simpler:
```ruby
regex_rules.none? { |regex_rule| regex_rule.check_for_match(entry) }
```
Or whatever.